### DISCUSSION TITLE
x-effect not cleaned up when the effect function interacts with the DOM

### DISCUSSION THREAD

**Discussion by bep** [alpinejs/alpine#2058] [7. Bugs (archived)]
See https://codepen.io/bepsays/pen/GREEmav

Note that I don't say that the below should work, but I had a head-scratching moment suggesting I should at least open this topic.

* If you click the blue button you see the console logging from the watcher
* If you click the red button, I expect the `x-effect` to go away, but it still logs to the console
* The "sinner" that seems to confuse Alpine/Vue is `this.$el.classList.add("foo")`, which I guess makes the dependency watch graph unintentionally big. The fix in this case was to bind the class the Alpine way, but this confused me pretty big.

Code pen content:

```js
<script>
  document.addEventListener('alpine:init', () => {
    Alpine.store('mystore', {
      counter1: 0,
      counter2: 0
    })
    Alpine.data('mydata', () => ({
      items: ["a", "b"],
      watchMe() {
        console.log("WATCH TRIGGERED ... COUNTER:", this.$store.mystore.counter1)
        // Comment out the line below and it starts behaving as expected.
        this.$el.classList.add("foo");
      }
    }))
  })
</script>
<div class="mx-auto container">
  <h1>Click 1</h1>
  <div id="c1" x-data="mydata" class="flex space-x-2" x-effect="watchMe()">
    <button class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded" @click="$store.mystore.counter1++" x-text="$store.mystore.counter1"></button>
  </div>

  <h1>Click 2</h1>
  <div id="c2" x-data="{ }" class="flex space-x-2">
    <button class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded" @click="$store.mystore.counter1++" x-text="$store.mystore.counter1"></button>
    <button class="bg-red-500 hover:bg-red-700 text-white font-bold py-2 px-4 rounded" @click="document.getElementById('c1').remove()">Remove Component 1</button>
  </div>
</div>
```

**Comment by SimoTod** [comment]
It seems a bug but I believe it's in the Vue reactivity layer since Alpine calls the expected clean up method.
It could be that it has been resolved in the latest version of vue/reactivity but I haven't tried it because Alpine uses 3.0 and 3.2 contains a breaking change so it's not fully compatible
